lib/types.nix

```
@@ -1107,6 +1107,48 @@ rec {
     addCheck = elemType: check: elemType // { check = x: elemType.check x && check x; };
 
   };
+
+  /**
+    Merges two option types together.
+
+    :::{.note}
+    Uses the merge function of the first type, to merge it with the second type.
```

This seems like an implementation detail

---

```suggestion
    Uses the type merge function of the first type, to merge it with the second type.
```

This is relevant for solving the "expression problem". Interactions between extensions don't have a natural home. This says that interactions are the responsibility of the first type. Concretely: if the types can be merged, you can only fix it if you control the first.
If you have to deal with a partially applied `mergeTypes`, you are subject to the provided type's merging support, with no way to extend its merging behavior.
If you control the first argument, you can add as much merge support as you like.

---

I had similar thoughts as roberth. If you want to change some of the merge behavior its important to know that merging is not commutative.